backend/flexyweek/settings.md
"""
Django settings for flexyweek project.

Generated by 'django-admin startproject' using Django 4.2.4.

For more information on this file, see
https://docs.djangoproject.com/en/4.2/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/4.2/ref/settings/
"""

from pathlib import Path

# Build paths inside the project like this: BASE_DIR / 'subdir'.
BASE_DIR = Path(__file__).resolve().parent.parent


# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/4.2/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = 'django-insecure-u5v&&voj^fys7ft)#30cj$w_*$&==v9qeo*ab2i^94z=ijwj&4'



# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True




# Application definition

INSTALLED_APPS = [
    'main',
    'rest_framework',
    'corsheaders',
    'knox',
    'authentication',
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
   
]

REST_FRAMEWORK = {
    'DEFAULT_AUTHENTICATION_CLASSES': (
         'knox.auth.TokenAuthentication',
    ),
}


ALLOWED_HOSTS = ['localhost', '127.0.0.1']


CORS_ALLOWED_ORIGINS = [
    'http://localhost:3000',
]


MIDDLEWARE = [
    'corsheaders.middleware.CorsMiddleware',  
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
]




ROOT_URLCONF = 'flexyweek.urls'

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
            ],
        },
    },
]

WSGI_APPLICATION = 'flexyweek.wsgi.application'


# Database
# https://docs.djangoproject.com/en/4.2/ref/settings/#databases

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.postgresql',
        'NAME': 'flexyweek_db',   # Nombre de tu base de datos en PostgreSQL
        'USER': 'postgres',                  # Nombre de usuario de PostgreSQL
        'PASSWORD': '',           # Contraseña de tu usuario de PostgreSQL
        'HOST': 'localhost',                   # Ruta del host de tu base de datos (por lo general, localhost)
        'PORT': '5432',                        # Puerto de tu base de datos (por defecto, 5432 para PostgreSQL)
    }
}


# Password validation
# https://docs.djangoproject.com/en/4.2/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
]


# Internationalization
# https://docs.djangoproject.com/en/4.2/topics/i18n/

LANGUAGE_CODE = 'es-es'  # Establece el idioma predeterminado en español (España)

TIME_ZONE = 'Europe/Madrid'  # Establece la zona horaria predeterminada en España (CET o CEST)

USE_I18N = True  # Habilita la internacionalización para admitir diferentes idiomas en tu aplicación

USE_L10N = True  # Habilita la localización para el formato de fechas, números y moneda según el idioma del usuario

USE_TZ = True  # Habilita el soporte de zona horaria, lo que permite trabajar con fechas y horas en diferentes zonas horarias


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/4.2/howto/static-files/

STATIC_URL = 'static/'

# Default primary key field type
# https://docs.djangoproject.com/en/4.2/ref/settings/#default-auto-field

DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'

# Configuración adicional para Celery
CELERY_BROKER_URL = 'redis://localhost:6379/0'  # URL de conexión a Redis
CELERY_RESULT_BACKEND = 'redis://localhost:6379/0'  # Opcional: configuración de Redis para resultados de tareas

backend/flexyweek/settings.py
"""
Django settings for flexyweek project.

Generated by 'django-admin startproject' using Django 4.2.4.

For more information on this file, see
https://docs.djangoproject.com/en/4.2/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/4.2/ref/settings/
"""

from pathlib import Path

# Build paths inside the project like this: BASE_DIR / 'subdir'.
BASE_DIR = Path(__file__).resolve().parent.parent


# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/4.2/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = 'django-insecure-u5v&&voj^fys7ft)#30cj$w_*$&==v9qeo*ab2i^94z=ijwj&4'

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True




# Application definition

INSTALLED_APPS = [
    'main',
    'rest_framework',
    'corsheaders',
    'knox',
    'authentication',
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
   
]

REST_FRAMEWORK = {
    'DEFAULT_AUTHENTICATION_CLASSES': (
         'knox.auth.TokenAuthentication',
    ),
}


ALLOWED_HOSTS = ['localhost', '127.0.0.1']


CORS_ALLOWED_ORIGINS = [
    'http://localhost:3000',
]


MIDDLEWARE = [
    'corsheaders.middleware.CorsMiddleware',  
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
]




ROOT_URLCONF = 'flexyweek.urls'

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
            ],
        },
    },
]

WSGI_APPLICATION = 'flexyweek.wsgi.application'


# Database
# https://docs.djangoproject.com/en/4.2/ref/settings/#databases

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.postgresql',
        'NAME': 'flexyweek_db',   # Nombre de tu base de datos en PostgreSQL
        'USER': 'postgres',                  # Nombre de usuario de PostgreSQL
        'PASSWORD': '',           # Contraseña de tu usuario de PostgreSQL
        'HOST': 'localhost',                   # Ruta del host de tu base de datos (por lo general, localhost)
        'PORT': '5432',                        # Puerto de tu base de datos (por defecto, 5432 para PostgreSQL)
    }
}


# Password validation
# https://docs.djangoproject.com/en/4.2/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
]


# Internationalization
# https://docs.djangoproject.com/en/4.2/topics/i18n/

LANGUAGE_CODE = 'es-es'  # Establece el idioma predeterminado en español (España)

TIME_ZONE = 'Europe/Madrid'  # Establece la zona horaria predeterminada en España (CET o CEST)

USE_I18N = True  # Habilita la internacionalización para admitir diferentes idiomas en tu aplicación

USE_L10N = True  # Habilita la localización para el formato de fechas, números y moneda según el idioma del usuario

USE_TZ = True  # Habilita el soporte de zona horaria, lo que permite trabajar con fechas y horas en diferentes zonas horarias


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/4.2/howto/static-files/

STATIC_URL = 'static/'

# Default primary key field type
# https://docs.djangoproject.com/en/4.2/ref/settings/#default-auto-field

DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'

# Configuración adicional para Celery
CELERY_BROKER_URL = 'redis://localhost:6379/0'  # URL de conexión a Redis
CELERY_RESULT_BACKEND = 'redis://localhost:6379/0'  # Opcional: configuración de Redis para resultados de tareas